ichnaea/api/views.py

```diff
@@ -79,9 +86,17 @@ def log_ip_and_rate_limited(self, valid_key, maxreq):
                 pipe.expire(log_ip_key, 691200)  # 8 days
                 pipe.incr(rate_key, 1)
                 pipe.expire(rate_key, 90000)  # 25 hours
-                _, _, count, _ = pipe.execute()
-                if maxreq and count > maxreq:
-                    should_limit = True
+                ip_count, _, limit_count, _ = pipe.execute()
+                log_params = {
+                    "api_key_count": limit_count,
+                    "api_key_ip_count": ip_count,
```

Comment:
I was wrong about [PFADD](https://redis.io/commands/pfadd). This is written as if it returns the uniqueness count, but instead it returns `1` if it has not seen this value (IP) before and adjusted the cardinality, and `0` if it has seen the value before, and the cardinality is the same.

I'm not sure it is useful to record if this is the first time this API/IP combo was seen today. Either the metric should be dropped, or the name changed.

Reply:
I'd like to keep this. I'd like a name that is short but self documenting. It would be a True/False value that says if this is the first time we've seen this IP address use this API key today, or if we've already seen this IP address use this API key today.

I did some brainstorming, here's some options sorted by length:

```
first
unique
first_ip
returning_ip
unique_by_ip
good_morning_ip
api_key_repeat_ip
api_key_returning_ip
first_for_key_and_ip
repeat_for_key_and_ip
api_key_first_ip_today
api_key_ip_first_today
api_key_repeat_customer
api_key_repeat_ip_today
first_for_key_and_ip_today
```

I think ``api_key_repeat_ip`` is a good compromise. It keeps it alpha-sorted with the rest of the API metrics, and communicates that it is a repeat IP for this API key. You'd have to dig to see that it starts counting daily, but I think a lot of the log metrics will be day-level anyway. Not perfect, open to inspiration.